File: dojo29/decode.js
const decode_me = "dX9ydEhnWwBsfQBEbHBBSkNHA2xeBxdHAEFO";
var b64_decoded = atob(decode_me).split("");
for (let pin = 0; pin < 10; pin++) {
  var decoded = [...b64_decoded];
  for (let i = 0; i < b64_decoded.length; i++) {
    decoded[i] = String.fromCharCode(
      b64_decoded[i].charCodeAt(0) ^ pin.toString().charCodeAt(0)
    );
  }
  console.log("Using pin " + pin + " : " + decoded.join(""));
}

File: dojo29/decode.py
#!/bin/python3
import base64

decrypt_me = "dX9ydEhnWwBsfQBEbHBBSkNHA2xeBxdHAEFO"
b64_decoded = str(base64.b64decode(decrypt_me).decode("utf-8"))
for pin in range(0,10):
    print(f"Using pin {pin} : ",end="")
    decoded = []
    for char in b64_decoded:
        c = ord(char) ^ ord(str(pin))
        decoded.append(chr(c))
    print("".join(decoded))

